File: app/lib/models/slot.dart
class Slot {
  final int id;
  final String date;
  final String startTime;
  final String endTime;
  final String timeSlot;
  final int maxCapacity;
  final int currentBookings;
  final int remainingCapacity;
  final bool isAvailable;

  Slot({
    required this.id,
    required this.date,
    required this.startTime,
    required this.endTime,
    required this.timeSlot,
    required this.maxCapacity,
    required this.currentBookings,
    required this.remainingCapacity,
    required this.isAvailable,
  });

  factory Slot.fromJson(Map<String, dynamic> json) {
    return Slot(
      id: json['id'] as int,
      date: json['date'] as String,
      startTime: json['start_time'] as String,
      endTime: json['end_time'] as String,
      timeSlot: json['time_slot'] as String,
      maxCapacity: json['max_capacity'] as int,
      currentBookings: json['current_bookings'] as int,
      remainingCapacity: json['remaining_capacity'] as int,
      isAvailable: json['is_available'] as bool,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'date': date,
      'start_time': startTime,
      'end_time': endTime,
      'time_slot': timeSlot,
      'max_capacity': maxCapacity,
      'current_bookings': currentBookings,
      'remaining_capacity': remainingCapacity,
      'is_available': isAvailable,
    };
  }

  bool get isAlmostFull => remainingCapacity <= 2;
}

File: api/app/Models/Slot.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Carbon\Carbon;

class Slot extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'date',
        'start_time',
        'end_time',
        'max_capacity',
        'current_bookings',
        'is_available',
    ];

    protected $casts = [
        'date' => 'date',
        'start_time' => 'datetime:H:i',
        'end_time' => 'datetime:H:i',
        'is_available' => 'boolean',
    ];

    protected $dates = [
        'deleted_at',
    ];

    // Check if slot is available for booking
    public function isBookable(): bool
    {
        return $this->is_available && 
               $this->current_bookings < $this->max_capacity &&
               $this->date >= Carbon::today();
    }

    // Get remaining capacity
    public function getRemainingCapacityAttribute(): int
    {
        return $this->max_capacity - $this->current_bookings;
    }

    // Format time slot display
    public function getTimeSlotAttribute(): string
    {
        return $this->start_time->format('H:i') . ' - ' . $this->end_time->format('H:i');
    }
}
